### PR TITLE
add function for constructing beanstalk client from URI

### PR DESCRIPTION
**Pull request by Roguelazer**
what it says on the tin

### CODE REVIEW

File: pystalk/client.py [hunk 1]
```diff
@@ -89,6 +92,21 @@ def __init__(self, host, port=11300, socket_timeout=None, auto_decode=False):
         self.desired_watchlist = set(['default'])
         self.auto_decode = auto_decode
 
+    @classmethod
+    def from_uri(cls, uri, socket_timeout=None, auto_decode=False):
+        parts = six.moves.urllib.parse.urlparse(uri)
+        ipv6_md = re.match(r'^\[([0-9a-fA-F:]+)\](:[0-9]+)?$', parts.netloc)
+        if ipv6_md:
+            host = ipv6_md.group(1)
+            port = ipv6_md.group(2) or 11300
```
Comment (jaredg) [inline pystalk/client.py]: Trim the colon and convert to integer?

File: tests/unit/test_pystalk.py [hunk 2]
```diff
@@ -28,3 +28,12 @@ def test_stats(self):
         self.server.responses.append(b'OK 17\r\n--- {"foo": "bar"}\r\n')
         assert self.client.stats() == {'foo': 'bar'}
         assert self.server.received == [b'stats\r\n']
+
+
+def test_from_uri():
+    client = pystalk.BeanstalkClient.from_uri('beanstalk://foo')
```
Comment (jaredg) [inline tests/unit/test_pystalk.py]: Test v4 and v6 addresses.

**Review by jaredg — CHANGES_REQUESTED** [review]

**Review by jaredg — APPROVED** [review]
New tests pass; since you have the other thing open to clean out stale versions I'm okay with it.

Thanks for humoring me.

**Review by victoryftw — COMMENTED** [review]
I see no issues, other than the CI test thing which doesn't seem blocking to me

**Review by victoryftw — APPROVED** [review]

**Pull request merged**